### PR TITLE
19.11 react jsx

### PR DESCRIPTION
**Pull request by pporc**

### CODE REVIEW

File: src/error/index.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export * from './error';
```
Comment (iliatcymbal) [inline src/error/index.js]: you file is `Error.js`, but you point out an `error` - why the names are different? It is not working for me.

File: src/header/greeting/Greeting.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	if (hours > 22 || hours > 0 && hours <= 3) { return 'Good night'; }
+
+	if (hours > 3 && hours < 12) { return 'Good morning'; }
+
+	if (hours >= 12 && hours < 18) { return 'Good afternoon'; }
+
+	return 'Good evening';
+}
+
+export const Greeting = (props) => (
```
Comment (iliatcymbal) [inline src/header/greeting/Greeting.js]: use destructuring `const Greeting = ({ name }) => (`

File: src/number/Number.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import './number.scss';
+
+const getAllNumbers = (...props) => {
+	const numbers = [];
+	const from = props[0];
+	const to = props[1];
+	const sort = props[2] ? 'odd' :
+				props[3] ? 'even' : false;
```
Comment (iliatcymbal) [inline src/number/Number.js]: props[0]
props[1]
props[2]

It is bad pattern. Use named arguments

File: src/number/Number.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	} else if (sort == 'odd') {
+		list = numbers.filter( num => num % 2)
+	} else {
+		list = numbers;
+	}
+	
+	return list;
+};
+
+export const Number = (props) => (
```
Comment (iliatcymbal) [inline src/number/Number.js]: use destructuring 
`const Number = ({ from, to, odd, even }) => (`

File: README.md [hunk 5]
```diff
@@ -1,4 +1,8 @@
-# React Todo App  
+# Todo application for the JS Advanced course 
 
 
-## Installation
+## How to setup
+
+1. Go to root directory and run _npm i_
+2. When setup is complated run _npm start_
+3. Open _dist/index.html_ to run application
```
Comment (iliatcymbal) [inline README.md]: page should be opened automatically

**Review by iliatcymbal — COMMENTED** [review]
You need to add next dependencies to your package.json:
- webpack-dev-server
- node-sass

Without it project isn't built

**Comment by pporc** [comment]
@iliatcymbal, fixed, review please.

File: src/number/Number.js [hunk 6]
```diff
@@ -1,12 +1,10 @@
...
 
-const getAllNumbers = (...props) => {
+const getAllNumbers = (from, to, odd, even) => {
 	const numbers = [];
-	const from = props[0];
-	const to = props[1];
-	const sort = props[2] ? 'odd' :
-				props[3] ? 'even' : false;
+	const sort = odd ? 'odd' :
+				even ? 'even' : false;
```
Comment (iliatcymbal) [inline src/number/Number.js]: it is too sophisticated...

```
let list = [];

	for (let i = from; i <= to; i++) {
		list.push(i);	
	}

	if (even) {
		list = list.filter( num => !(num % 2))
	}
	
	if (odd) {
		list = list.filter( num => num % 2)
	} 
	
	// if (even && odd) ---> there you can define what to do in this case (if needed)
	
	return list;
```

**Review by iliatcymbal — APPROVED** [review]
Fix one note and merge to master

**Pull request merged**